src/main/java/org/scoverage/plugin/SCoveragePreCompileMojo.java

```diff
@@ -457,9 +457,20 @@ private Artifact getScalaScoveragePluginArtifact( String scalaMainVersion )
             }
         }
 
-        result =
-            getResolvedArtifact( "org.scoverage", "scalac-scoverage-plugin_" + scalaMainVersion,
-                                 resolvedScalacPluginVersion );
+        try
+        {
+            // Look for plugin artifact matching the scala version (full form like 2.12.14)
+            // If not found then look for artifact based on major version like 2.12
+            result =
+                    getResolvedArtifact( "org.scoverage", "scalac-scoverage-plugin_" + resolvedScalaVersion,
+                            resolvedScalacPluginVersion );
+        } catch (ArtifactNotFoundException e)
```

Comment:
If an exception were thrown from both the above try-block the below catch-block caught, then the error message will only report not finding the `scalaMainVersion` (rather than both the `scalaMainVersion` and `resolvedScalaVersion`).

(I could be wrong though. Even if my comment is correct I don't whether a slightly inaccurate error message would be considered a good reason to block the pull request.)